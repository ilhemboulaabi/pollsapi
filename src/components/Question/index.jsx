import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  Card,
  CardMedia,
  CardContent,
  Typography,
  Grid,
  CardActions,
  Button,
  LinearProgress,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import Moment from 'moment';
import { Link } from 'react-router-dom';

const useStyles = makeStyles(() => ({
  card: {
    display: 'flex',
  },
  container: {
    display: 'flex',
    flexDirection: 'column',
    width: '60%',
  },
  media: {
    width: 140,
  },
}));

const Question = ({ question, loading }) => {
  const classes = useStyles();

  return (
    loading
      ? <LinearProgress />
      : (
        <Grid item xs={12}>
          <Card className={classes.card} raised>
            <CardMedia
              className={classes.media}
              image="https://cdn.pixabay.com/photo/2015/11/03/08/56/question-mark-1019820_960_720.jpg"
              title="Contemplative Reptile"
            />
            <div className={classes.container}>
              <CardContent>
                <Typography component="h5" variant="h5">
                  {question.question}
                </Typography>
                <Typography variant="subtitle1" color="textSecondary">
                  {Moment(question.published_at).format('d MMM yyyy')}
                </Typography>
              </CardContent>
              <CardActions>
                <Button size="small" color="primary">
                  <Link to={`${question.url}`}>Vote here!</Link>
                </Button>
              </CardActions>
            </div>
          </Card>
        </Grid>
      )
  );
};

export default Question;

Question.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  question: PropTypes.object,
  loading: PropTypes.bool,
};
Question.defaultProps = {
  question: {},
  loading: false,
};
