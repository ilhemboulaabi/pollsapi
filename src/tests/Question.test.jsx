import * as React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Question from '../components/Question';
import questionData from './mockData/data';

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    question: questionData,
    loading: false,
  };

  const wrapper = shallow(<Question {...props} />);

  return {
    props,
    wrapper,
  };
};

describe('Question', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
