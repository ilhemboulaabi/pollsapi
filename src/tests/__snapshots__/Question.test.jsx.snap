// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render self 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
    raised={true}
  >
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-3"
      image="https://cdn.pixabay.com/photo/2015/11/03/08/56/question-mark-1019820_960_720.jpg"
      title="Contemplative Reptile"
    />
    <div
      className="makeStyles-container-2"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h5"
          variant="h5"
        >
          Favourite programming language?
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          variant="subtitle1"
        >
          2 Nov 2014
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          size="small"
        >
          <Link
            to="/questions/1"
          >
            Vote here!
          </Link>
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </div>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Grid))>
`;
